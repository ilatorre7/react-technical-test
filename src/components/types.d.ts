export interface IItem {
  id: number,
  title: string,
  image: string,
  description: string,
  detail: {
    brand: string,
    price: string,
    year: number,
    colorway: string
  }
}

export interface IOption { 
  label: string, 
  value: string 
}