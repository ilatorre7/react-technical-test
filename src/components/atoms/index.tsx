import { InputSelect } from "./InputSelect";
import { Button } from "./Button";

export { InputSelect, Button }