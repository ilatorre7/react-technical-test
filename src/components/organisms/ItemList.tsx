import "../../styles/ItemList.css"
import { IItem } from "../types"
import { ItemCard } from "../molecules/ItemCard"

export const ItemList = ({ list, isLoading }: { list: IItem[], isLoading: boolean }) => {

  return (
    <div className='list-container'>
      {
        list.map((item) => (
          <ItemCard key={`item-${item.id}`} {...item} />
        ))
      }
      {isLoading && <p>Loading...</p>}
    </div>
  )
}