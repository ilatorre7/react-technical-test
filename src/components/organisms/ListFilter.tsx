import '../../styles/Aside.css'
import { IOption } from "../types"
import { InputSelect } from "../atoms/InputSelect"

export const ListFilter = ({ options, onFilter } : { options: IOption[], onFilter : (e: string) => void } ) => {

  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    e.preventDefault()
    onFilter(e.target.value)
  }

  return (
    <aside className='aside'>
      <h3>Filtrar por:</h3>
      <InputSelect placeholder='Selecciona una marca para filtrar' options={options} onChange={handleChange} />
    </aside>
  )
}