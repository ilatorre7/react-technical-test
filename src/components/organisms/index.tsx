import { ItemList } from "./ItemList";
import { ListFilter } from "./ListFilter";

export { ItemList, ListFilter }