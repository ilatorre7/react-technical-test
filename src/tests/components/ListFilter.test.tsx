import { describe, it, expect } from 'vitest'
import { render } from '@testing-library/react';
import { ListFilter } from '../../components/organisms';

describe('<ItemList />', () => {
  const options = [
    { label: "Jordan", value: "Jordan" },
    { label: "Nike", value: "Nike"},
    { label: "Adidas", value: "Adidas"},
    { label: "Puma", value: "Puma"},
    { label: "Reebok", value: "Reebok"},
    { label: "Converse", value: "Converse"},
    { label: "Vans", value: "Vans"},
    { label: "Fila", value: "Fila"},
    { label: "New Balance", value: "New Balance"},
    { label: "Mizuno", value: "Mizuno"},
    { label: 'Asics', value: 'Asics'}
  ]

  it('Should render a list of cards with the correct values', () => {
    const { container } = render(<ListFilter options={options} onFilter={() => {}} />)
    expect(container).toMatchSnapshot()
  })
})