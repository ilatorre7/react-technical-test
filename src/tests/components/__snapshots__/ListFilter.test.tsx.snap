// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ItemList /> > Should render a list of cards with the correct values 1`] = `
<div>
  <aside
    class="aside"
  >
    <h3>
      Filtrar por:
    </h3>
    <select
      class="card-filter"
      name="filter"
    >
      <option
        value=""
      >
        Selecciona una marca para filtrar
      </option>
      <option
        value="Jordan"
      >
        Jordan
      </option>
      <option
        value="Nike"
      >
        Nike
      </option>
      <option
        value="Adidas"
      >
        Adidas
      </option>
      <option
        value="Puma"
      >
        Puma
      </option>
      <option
        value="Reebok"
      >
        Reebok
      </option>
      <option
        value="Converse"
      >
        Converse
      </option>
      <option
        value="Vans"
      >
        Vans
      </option>
      <option
        value="Fila"
      >
        Fila
      </option>
      <option
        value="New Balance"
      >
        New Balance
      </option>
      <option
        value="Mizuno"
      >
        Mizuno
      </option>
      <option
        value="Asics"
      >
        Asics
      </option>
    </select>
  </aside>
</div>
`;
